### ISSUE TITLE
react native: open error

### ISSUE THREAD

**Issue by ptScaler** [xmppjs/xmpp.js#761]
```
"@xmpp/client": "^0.7.4",
```

```
status: connecting 
status: connect 
status: opening 
status: open <open xmlns="urn:ietf:params:xml:ns:xmpp-framing" id="14170668206454173697" version="1.0" xml:lang="en" from="xmpp.m2os.cn"/>

undefined is not an object (evaluating 'Base64.btoa')
- node_modules/@xmpp/sasl/lib/b64.js:14:16 in encode
- node_modules/@xmpp/sasl/index.js:69:17 in <unknown>
- ... 88 more stack frames from framework internals

❌, TypeError: undefined is not an object (evaluating 'Base64.btoa')
- node_modules/expo/build/environment/muteWarnings.fx.js:26:24 in error
* screens/ChatScreen.js:72:26 in <unknown>
- node_modules/events/events.js:151:19 in emit
- node_modules/promise/setimmediate/core.js:37:14 in tryCallOne
- ... 9 more stack frames from framework internals
```

**Comment by sonnyp** [comment]
Strange, RN cannot find `js-base64`. Which version of RN are you using?

**Comment by ptScaler** [comment]
Using the latest V0.60， in node_modules also find  js-base 64, it is not clear why this error was reported.

**Comment by ptScaler** [comment]
Check previous issue solution。
```
import {Base64} from 'js-base64';
global.btoa = Base64.encode;
global.atob = Base64.decode;
```

**Comment by sonnyp** [comment]
@ptScaler are you refering to a previous issue? Can you share the link?

Thanks

**Comment by ptScaler** [comment]
#536 I found a solution here

**Comment by ptScaler** [comment]
@sonnyp Excuse me， I found that the following code was not needed in the react-native debug state，But what if you had to have that in order for the release to work。
```
import {Base64} from 'js-base64';
global.btoa = Base64.encode;
global.atob = Base64.decode;
```

**Comment by sonnyp** [comment]
@ptScaler that is strange.

Which RN version are you using? Is it ejected?

**Comment by ptScaler** [comment]
```
"version": "0.59.8"
```
yeah, my app is ejected. in dubug state，I add the above code and throw an exception, linking out of time，However, in the release version, I could not see the log, and found that only the above code could work properly if it was opened
```
StreamError: connection-timeout - Idle connection
ChatListScreen.js:79 error
ChatListScreen.js:80 StreamError: connection-timeout - Idle connection
    at Function.fromElement (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:132854:21)
    at Client._onElement (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:130551:40)
    at FramedParser.listeners.element (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:130564:18)
    at FramedParser.emit (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:131197:7)
    at FramedParser.onEndElement (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:133271:16)
    at SaxLtx.emit (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:131197:7)
    at SaxLtx._handleTagOpening (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:132537:14)
    at SaxLtx.write (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:132662:22)
    at FramedParser.write (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:132479:21)
    at Client._onData$ (blob:http://localhost:8081/23e4902a-3040-4057-ae90-7710c0a25679:130478:62)
RCTLog.js:47 SocketRocket: In debug mode.  Allowing connection to any root cert
myxmpp.js:22 RangeError: Maximum call stack size exceeded
ChatListScreen.js:79 error
ChatListScreen.js:80 RangeError: Maximum call stack size exceeded
    at String.replace (<anonymous>)
    at utob (base64.js:66)
    at _encode (base64.js:96)
    at encode (base64.js:100)
    at btoa (base64.js:82)
    at _encode (base64.js:96)
    at encode (base64.js:100)
    at btoa (base64.js:82)
    at _encode (base64.js:96)
    at encode (base64.js:100)
```

**Comment by sonnyp** [comment]
fixed in 0.9.0 https://github.com/xmppjs/xmpp.js/releases/tag/v0.9.0
